crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -30,6 +30,17 @@ def node_status
         }
       end
 
+      def upgrade_mode
+          ::Crowbar::UpgradeStatus.new.upgrade_mode
+      end
+
+      def upgrade_mode=(mode)
+        unless ["normal", "non_disruptive"].include? mode
+          Rails.logger.debug("Setting upgrade mode #{mode}")
+          ::Crowbar::UpgradeStatus.new.save_selected_upgrade_mode(mode.to_sym)
+        end
```

Comment:
What if user passes different value? There should be some help message showing allowed values in such case...